meteortracker/save_event.py

```diff
@@ -0,0 +1,126 @@
+
+    def __del__(self):
+        self.conn.close()
+
+    def add_event(self, current_image, previous_image):
+        date = datetime.datetime.utcnow().isoformat()
+        latitude = self.config.get('Location', 'Latitude', 0)
+        longitude = self.config.get('Location', 'Longitude', 0)
+        bearing = self.config.get('Location', 'Bearing', 0)
+        roll = self.config.get('Location', 'Roll', 0)
+        pitch = self.config.get('Location', 'Pitch', 0)
+        yaw = self.config.get('Location', 'Yaw', 0)
+
+        intrinsic_matrix = self.config.get('Camera', 'IntrinsicMat', 0)
+        distortion_coefficient = self.config.get('Camera', 'DistortionCoeff', 0)
+
+        if self.use_local:
+            self.upload_to_local(
+                current_image, previous_image, date, latitude, longitude,
+                bearing, roll, pitch, yaw,
+                intrinsic_matrix, distortion_coefficient
+            )
+        if self.use_remote:
+            self.upload_to_remote(
+                current_image, previous_image, date, latitude, longitude,
+                bearing, roll, pitch, yaw,
+                intrinsic_matrix, distortion_coefficient
+            )
+
+    def upload_to_remote(self, current_image, previous_image, date,
+                         latitude, longitude, bearing,
+                         roll, pitch, yaw,
+                         intrinsic_matrix, distortion_coefficient):
+        ...
+
+    def upload_to_local(self, current_image, previous_image, date,
+                        latitude, longitude, bearing,
+                        roll, pitch, yaw,
+                        intrinsic_matrix, distortion_coefficient):
+        # save the images locally, we don't want them in the database so they're
+        # easier to work with
+        filename_format = '{location}_{date}_{{order}}.jpg'.format(
+            location=self.local_image_location, date=date)
+
+        filename_current = filename_format.format(order='current')
+        filename_previous = filename_format.format(order='previous')
+
+        cv2.imwrite(filename_current, current_image)
+        cv2.imwrite(filename_previous, previous_image)
+
+        sql_template = 'insert into {table_name} ({fields}) values ({values})'
+        db_command = sql_template.format(
+            table_name=self.local_db_table_name,
+            fields=", ".join(variables),
+            values=", ".join(str(_read(variable))
+                             for variable in self._variables)
+        )
+        self.conn.execute(db_command)
+        self.conn.commit()
+
+    def check_local_db(self):
+        sql_template = 'create table if not exists {table_name} ({fields})'
+        db_command = sql_template.format(
+            table_name = self.local_db_table_name,
+            fields=", ".join(
+                variable
+                + (" REAL" if isinstance(_read(variable), numbers.Real) else "")
```

Comment:
Use a dict instead of a tuple.

    variables = {
        "latitude": "REAL",
        "longitude": "REAL",
    }

What happens to the variables that don't have a type specified?